Lib/test/test_pyclbr.py

```diff
@@ -22,6 +26,27 @@
 # is imperfect (as designed), testModule is called with a set of
 # members to ignore.
 
+
+@contextmanager
+def temporary_main_spec():
+    """
+     A context manager that temporarily sets the `__spec__` attribute
+    of the `__main__` module if it's missing.
+    """
+    original_spec = getattr(sys.modules["__main__"], "__spec__", None)
```

Comment:
Is it possible for `__main__` not to be in `sys.modules`? if `__main__` is not present, let's just yield without doing anything.